StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1027UseTabsCorrectly.cs

```diff
@@ -0,0 +1,268 @@
+        {
+            ImmutableArray<TextSpan>.Builder builder = ImmutableArray.CreateBuilder<TextSpan>();
+
+            // Locate disabled text
+            foreach (var trivia in root.DescendantTrivia(descendIntoTrivia: true))
+            {
+                if (trivia.IsKind(SyntaxKind.DisabledTextTrivia))
+                {
+                    builder.Add(trivia.Span);
+                }
+                else if (trivia.IsKind(SyntaxKind.SingleLineCommentTrivia))
+                {
+                    if (trivia.ToString().StartsWith("////"))
+                    {
+                        // Exclude comments starting with //// because they could contain commented code which contains
+                        // string or character literals, and we don't want to change the contents of those strings.
+                        builder.Add(trivia.Span);
+                    }
+                }
+            }
+
+            // Locate string literals
+            foreach (var token in root.DescendantTokens(descendIntoTrivia: true))
+            {
+                switch (token.Kind())
+                {
+                case SyntaxKind.XmlTextLiteralToken:
+                    if (token.Parent.IsKind(SyntaxKind.XmlCDataSection))
+                    {
+                        builder.Add(token.Span);
+                    }
+
+                    break;
+
+                case SyntaxKind.CharacterLiteralToken:
+                case SyntaxKind.StringLiteralToken:
+                case SyntaxKind.InterpolatedStringTextToken:
+                    builder.Add(token.Span);
+                    break;
+
+                default:
+                    break;
+                }
+            }
+
+            // Sort the results
+            builder.Sort();
+
+            // Combine adjacent and overlapping spans
+            ReduceTextSpans(builder);
+
+            excludedSpans = builder.ToImmutable();
+            return true;
+        }
+
+        private static void ReduceTextSpans(ImmutableArray<TextSpan>.Builder sortedTextSpans)
+        {
+            if (sortedTextSpans.Count == 0)
+            {
+                return;
+            }
+
+            int currentIndex = 0;
+            for (int nextIndex = 1; nextIndex < sortedTextSpans.Count; nextIndex++)
+            {
+                TextSpan current = sortedTextSpans[currentIndex];
+                TextSpan next = sortedTextSpans[nextIndex];
+                if (current.End < next.Start)
+                {
+                    // Increment currentIndex this iteration
+                    currentIndex++;
+
+                    // Only increment nextIndex this iteration if necessary to ensure nextIndex > currentIndex on the
+                    // next iteration. At this point we already incremented currentIndex, but haven't incremented
+                    // nextIndex.
+                    if (currentIndex > nextIndex)
```

Comment:
I think it's worse than that actually... seems to be missing the following after incrementing `currentIndex`:

```csharp
sortedTextSpans[currentIndex] = next;
```